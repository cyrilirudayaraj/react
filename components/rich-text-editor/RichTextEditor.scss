.bg-color-picker img,
.text-color-picker img {
  width: 15px;
  height: 15px;
}
.rdw-colorpicker-modal .bgcolor {
  margin-left: 35px;
}

.rdw-colorpicker-modal button {
  border: 1px solid #edebe9;
  box-sizing: border-box;
  border-radius: 3px;
  text-align: center;
  width: 100%;
  min-height: 2rem;
  background: 0;
}
.rdw-colorpicker-modal {
  width: 178px;
  height: 149px;
}

// Rich text editor
.rte {
  .fe_c_form-field__input-slot {
    padding-right: 31px;
  }
  .editor-wrapper {
    overflow: hidden !important;
    height: 250px;
  }
  .editor-wrapper .editor-textarea {
    height: 94%;
  }
  .DraftEditor-root {
    z-index: 0;
  }
  .editor-textarea::-webkit-scrollbar {
    width: 5px;
  }
  .editor-textarea::-webkit-scrollbar-thumb {
    border-radius: 10px;
  }
  .rdw-colorpicker-modal-options {
    overflow: hidden;
  }
  .rdw-colorpicker-modal-options::-webkit-scrollbar {
    width: 5px;
  }
  .rdw-colorpicker-modal-options::-webkit-scrollbar-thumb {
    border-radius: 10px;
  }
  .rdw-dropdown-optionwrapper::-webkit-scrollbar {
    width: 5px;
  }
  .rdw-dropdown-optionwrapper::-webkit-scrollbar-thumb {
    border-radius: 10px;
  }

  .discussion_text_box::-webkit-scrollbar {
    width: 5px;
  }
  .discussion_text_box::-webkit-scrollbar-thumb {
    border-radius: 10px;
  }

  .discussion_text_box {
    background: white;
    border: 1px solid #b3b3b3;
    border-radius: 3px;
    height: 250px;
    min-height: 250px;
    max-height: 250px;
    overflow: hidden !important;
    width: 115%;
    resize: none;
    transition: box-shadow 250ms cubic-bezier(0.215, 0.61, 0.355, 1),
      border-color 250ms cubic-bezier(0.215, 0.61, 0.355, 1),
      background-color 250ms cubic-bezier(0.215, 0.61, 0.355, 1),
      color 250ms cubic-bezier(0.215, 0.61, 0.355, 1),
      fill 250ms cubic-bezier(0.215, 0.61, 0.355, 1),
      stroke 250ms cubic-bezier(0.215, 0.61, 0.355, 1);
  }

  .discussion_text_box .editor-textarea {
    height: 88%;
    //overflow-y: scroll;
  }

  .discussion_text_box:hover {
    border: 1px solid #0466b4;
    box-shadow: 0 1px 8px 0 rgba(0, 0, 0, 0.25);
  }

  .discussion_text_box:focus {
    outline: none;
  }
  .rdw-link-modal {
    height: inherit;
  }

  .rdw-link-decorator-wrapper a span:hover {
    cursor: text;
  }

  .rdw-link-decorator-icon {
    cursor: pointer;
    position: right;
  }

  .rdw-right-aligned-block {
    text-align: right !important;
  }

  .rdw-link-modal-target-option {
    display: none;
  }

  .rdw-option-active {
    background: #d9ebfa !important;
    border: 1px solid #0275d4 !important;
    color: #0275d4;
    box-shadow: 0 2px 8px rgba(2, 117, 212, 0.2);
  }

  .rdw-dropdown-wrapper,
  .rdw-option-wrapper {
    border: 1px solid #b3b3b3;
    background: #ffffff;
    box-sizing: border-box;
    border-radius: 3px;
    height: 28px;
  }

  .rdw-dropdown-optionwrapper {
    margin: 2px 0;
  }
  .rdw-dropdown-wrapper:hover,
  .rdw-option-wrapper:hover {
    border: 1px solid #0275d4;
    cursor: pointer;
    border-color: #0275d4;
    color: #0275d4;
    box-shadow: 0 2px 8px rgba(2, 117, 212, 0.2);
  }

  .rdw-dropdown-wrapper a {
    border: none;
    color: black;
  }

  .rdw-colorpicker-option {
    border: none;
    height: 22px;
  }

  .rdw-colorpicker-option:hover {
    border: none;
  }

  .rdw-link-decorator-icon {
    right: 0px;
    left: unset;
    top: -9px;
  }
}
