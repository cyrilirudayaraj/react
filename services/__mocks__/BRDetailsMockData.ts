import { BRDetails } from '../../types';

export const BRDetailsMockData1: BRDetails = {
  ruleReportingCategoryId: '9',
  rules2TransformationStatus: '',
  internalFixText2:
    '<h2 style="text-align:left;"><span style="color: rgb(0,0,0);background-color: rgb(255,255,255);font-size: 24px;font-family: DauphinPlain;">Where can I get some?</span></h2>\n<p style="text-align:justify;"><span style="color: rgb(0,0,0);background-color: rgb(255,255,255);font-size: 14px;font-family: Open Sans", Arial, sans-serif;">There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don\'t look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn\'t anything embarrassing hidden in the middle of text. All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum which looks reasonable. The generated Lorem Ipsum is therefore always free from repetition, injected humour, or non-characteristic words etc.</span>&nbsp;</p>',
  ruleTypeId: '3',
  internalFixText1:
    '<p><strong>Lorem Ipsum</strong> <span style="color: rgb(0,0,0);background-color: rgb(255,255,255);font-size: 14px;font-family: Open Sans;">is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.</span>&nbsp;</p>',
  visitRuleDisplayLocation: 'VISITBILL_MEDICALCODES',
  contextName: 'CA - Prime Care - Inland Valley',
  contextId: '21',
  id: '1251',
  visitRuleDisplayLocationId: '2',
  businessRequirementTypeId: '2',
  legacyRuleId: '',
  rules2TransformationStatusId: '',
  name: 'Business Requirement Name',
  ruleReportingCategoryName: 'Coding: Procedure Code',
  description:
    '<p><span style="color: rgb(95,95,95);background-color: rgb(255,255,255);font-size: 14px;font-family: Source Sans Pro", arial, sans-serif;">Business Requirement Description (Client Friendly)</span>&nbsp;</p>',
  ruleType: 'CLAIMCREATE',
  localRuleUseCaseId: '2',
  localRuleUseCaseName: 'Capitation',
  businessRequirementType: 'Local',
  associatedBRTasks: [
    {
      id: '12',
      taskName: 'test1',
      statusId: '2',
      priorityId: '1',
      statusName: 'In Progress',
      escalatedYn: 'N',
      deploymentDate: '07/28/2021',
    },
    {
      id: '13',
      taskName: 'test2',
      statusId: '3',
      priorityId: '3',
      statusName: 'Assigned',
      escalatedYn: 'Y',
      deploymentDate: '07/29/2021',
    },
  ],
};
