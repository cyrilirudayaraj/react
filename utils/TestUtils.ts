//TODO exclude this file during bundling
export function findByName(wrapper: any, name: string): any {
  const selector = "input[name='" + name + "']";
  return wrapper.find(selector).last();
}

export function findById(wrapper: any, id: string): any {
  const selector = '#' + id;
  return wrapper.find(selector).last();
}

export function changeValueById(wrapper: any, id: string, value: string): void {
  findById(wrapper, id).simulate('change', {
    target: { name: id, value: value },
  });
}

export function sleep(x: number): Promise<any> {
  return new Promise<void>((resolve) => {
    setTimeout(() => {
      resolve();
    }, x);
  });
}
