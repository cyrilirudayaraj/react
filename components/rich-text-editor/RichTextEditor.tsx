import { FormField } from '@athena/forge';
import { ContentState, convertToRaw, EditorState } from 'draft-js';
import draftToHtml from 'draftjs-to-html';
import { FormikProps } from 'formik';
import htmlToDraft from 'html-to-draftjs';
import { get } from 'lodash';
import React from 'react';
import { Editor } from 'react-draft-wysiwyg';
import Labels from '../../constants/Labels';
import '../../../node_modules/react-draft-wysiwyg/dist/react-draft-wysiwyg.css';
import { connect } from 'react-redux';
import { ColorPicker } from './ColorPicker';
import './RichTextEditor.scss';
import { getUsers, fetchUsersOnce } from '../../slices/MasterDataSlice';

interface RichTextEditorProps {
  id: string;
  wrapperId: number;
  labelText?: string;
  formik: FormikProps<any>;
  isDisabled?: boolean;
  isRequired?: boolean;
  maxLength: number;
  className?: string;
  enableShowMore?: boolean;
  fetchUsersOnce?: any;
  usersInfo: any[];
  enableTagging?: boolean;
}

interface RichTextEditorState {
  content: string;
  editorState: EditorState;
  hideToolBar: boolean;
  bordercolor: string;
  showCharCount: boolean;
  enableShowMore: boolean;
}

export class RichTextEditor extends React.Component<
  RichTextEditorProps,
  RichTextEditorState
> {
  HEIGHT = '250px';
  SHOW_MORE_MAX_HEIGHT = '474px';

  static defaultProps = {
    labelText: '',
    maxLength: 4000,
    isRequired: false,
    className: 'fe_c_textarea fe_c_textarea--auto-size editor-wrapper',
    enableShowMore: true,
  };

  constructor(props: RichTextEditorProps) {
    super(props);
    this.props.fetchUsersOnce();
  }

  state = {
    content: '',
    editorState: EditorState.createEmpty(),
    hideToolBar: true,
    bordercolor: '',
    showCharCount: false,
    enableShowMore: this.props.enableShowMore || false,
  };

  setEditorToNormalHeight = (): void => {
    const targetObject = this.getEditorElement();
    if (targetObject !== null) {
      targetObject.style.maxHeight = this.HEIGHT;
    }
  };

  enableCharCount = (showCharcount: boolean): void => {
    this.setState({
      showCharCount: showCharcount,
    });
  };

  getEditorElement = (): HTMLElement | null => {
    return document.getElementById('rdw-wrapper-' + this.props.wrapperId);
  };

  toggleExpand = (event: any): void => {
    const buttonelement = document.getElementById(event.target.id);
    const textelement = this.getEditorElement();
    if (buttonelement && textelement) {
      if (buttonelement.innerText === Labels.COMPONENTS.TEXTAREA.SHOW_MORE) {
        textelement.style.height = this.SHOW_MORE_MAX_HEIGHT;
        textelement.style.maxHeight = this.SHOW_MORE_MAX_HEIGHT;
        buttonelement.innerText = Labels.COMPONENTS.TEXTAREA.SHOW_LESS;
      } else {
        textelement.style.height = this.HEIGHT;
        textelement.style.maxHeight = this.HEIGHT;
        buttonelement.innerText = Labels.COMPONENTS.TEXTAREA.SHOW_MORE;
      }
    }
  };

  getBorderColor = (): string => {
    return this.getPlainText().length > this.props.maxLength ? '#FFAB00' : '';
  };
  onFocus = () => {
    this.enableCharCount(true);
    this.setState({ hideToolBar: false });
  };
  onBlur = () => {
    this.setToFormikProps();
    this.enableCharCount(false);
    this.setEditorToNormalHeight();
    this.setState({ hideToolBar: true });
  };

  getPlainText() {
    return this.state.editorState.getCurrentContent().getPlainText();
  }

  onEditorStateChange = (editorState: EditorState) => {
    this.setState({
      editorState,
      bordercolor: this.getBorderColor(),
    });
  };

  toolBarOptions: any = {
    options: [
      'inline',
      'fontSize',
      'colorPicker',
      'link',
      'list',
      'textAlign',
      'history',
    ],
    colorPicker: {
      component: ColorPicker,
      colors: [
        '#E25041',
        '#FBA026',
        '#F7DA64',
        '#41A85F',
        '#2C82C9',
        '#9365B8',
        '#D1D5D8',
        '#A38F84',
        '#000000',
        '#FFFFFF',
      ],
    },
    inline: {
      options: ['bold', 'italic', 'underline', 'strikethrough'],
    },
    link: {
      options: ['link'],
      defaultTargetOption: '_blank',
    },
  };

  showCharCount() {
    return (
      this.state.showCharCount && (
        <span className="charcount">
          {this.getPlainText().length}/{this.props.maxLength}
        </span>
      )
    );
  }

  setToFormikProps() {
    const formikvalue = this.props.formik.values[this.props.id];
    const currentvalue = draftToHtml(
      convertToRaw(this.state.editorState.getCurrentContent())
    );
    if (formikvalue != currentvalue) {
      this.props.formik.setFieldValue(
        this.props.id,
        currentvalue.replace(/[^\x00-\x7F]/g, '')
      );
    }
  }

  getErrorFromFormik(): string | undefined {
    return get(this.props.formik.errors, this.props.id)?.toLocaleString();
  }

  showMoreLess(): React.ReactNode {
    // If showmoreless is enabled and charcount is false display show more or less
    return (
      this.state.enableShowMore &&
      this.state.showCharCount === false && (
        <div className="counter">
          <a
            id={'showmoreless-' + this.props.id}
            onClick={(event: any) => this.toggleExpand(event)}
          >
            {Labels.COMPONENTS.TEXTAREA.SHOW_MORE}
          </a>
        </div>
      )
    );
  }

  checkShowMore = (event: any): void => {
    const targetObject = this.getEditorElement();
    if (targetObject?.style) {
      if (targetObject?.style.height > this.HEIGHT) {
        this.setState({
          enableShowMore: true,
        });
      } else {
        this.setState({
          enableShowMore: false,
        });
      }
    }
  };

  reinitializeStateContent() {
    const fieldValue = get(this.props.formik.values, this.props.id, '') || '';
    if (this.state.content != fieldValue) {
      const editorState = EditorState.createWithContent(
        ContentState.createFromBlockArray(
          htmlToDraft(fieldValue).contentBlocks,
          htmlToDraft(fieldValue).entityMap
        )
      );
      this.setState({
        content: fieldValue,
        editorState: editorState,
      });
    }
  }

  render() {
    const wrapperClassName = this.addClassName();
    this.reinitializeStateContent();
    return (
      <div className="textarea-component">
        <FormField
          id={this.props.id}
          // This wrapperId has to be provided or else its autogenerated which is causing snapshot failing
          wrapperId={this.props.wrapperId}
          labelText={this.props.labelText}
          labelWidth={1}
          className="rte"
          inputAs={Editor}
          required
          wrapperClassName={wrapperClassName}
          editorClassName="editor-textarea"
          editorState={this.state.editorState}
          onEditorStateChange={this.onEditorStateChange}
          readOnly={this.props.isDisabled}
          wrapperStyle={{ borderColor: this.state.bordercolor }}
          toolbarHidden={this.state.hideToolBar}
          error={this.getErrorFromFormik()}
          toolbar={this.toolBarOptions}
          onChange={this.checkShowMore}
          onFocus={this.onFocus}
          onBlur={this.onBlur}
          {...(this.props.enableTagging && {
            mention: {
              separator: ' ',
              trigger: '@',
              suggestions: this.props.usersInfo,
            },
          })}
        ></FormField>
        {this.showMoreLess()}
        {this.showCharCount()}
      </div>
    );
  }
  addClassName() {
    let wrapperClassName = this.props.className;

    const error = get(
      this.props.formik.errors,
      this.props.id
    )?.toLocaleString();

    if (this.props.isRequired) {
      wrapperClassName += ' fe_is-required';
    }
    if (error !== undefined) {
      wrapperClassName += ' fe_is-error';
    }

    if (this.props.isDisabled) {
      wrapperClassName += ' fe_is-disabled';
    }
    return wrapperClassName;
  }
}
const constructSuggestions = (usersInfo: any): any[] => {
  return usersInfo.map((userinfo: any) => {
    return {
      text: userinfo.firstName + ' ' + userinfo.lastName,
      value: userinfo.userName,
      url: process.env.REACT_APP_PROFILE_URL + userinfo.userName,
    };
  });
};
const mapStateToProps = (state: any) => {
  return {
    usersInfo: constructSuggestions(getUsers(state)),
  };
};

export default connect(mapStateToProps, { fetchUsersOnce })(RichTextEditor);
