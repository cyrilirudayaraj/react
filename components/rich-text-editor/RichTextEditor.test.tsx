import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import RichTextEditor from './RichTextEditor';
import { findById } from '../../utils/TestUtils';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { MockData } from '../../services/__mocks__/MockData';

configure({ adapter: new Adapter() });
jest.mock('draft-js/lib/generateRandomKey', () => () => '123');

describe('test <RichTextEditor>', () => {
  let wrapper: any;
  const id = 'description';
  const wrapperId = 1;
  const eleWrapperId = 'rdw-wrapper-' + wrapperId;
  const formik: any = {
    values: { description: 'Testing value' },
    touched: {},
    errors: {},
    setFieldValue: jest.fn(),
  };

  beforeEach(() => {
    const store = configureStore([])({
      masterData: {
        users: MockData.USERS,
      },
      fetchUsersInfoOnce: jest.fn(),
    });
    store.dispatch = jest.fn();
    wrapper = mount(
      <Provider store={store}>
        <RichTextEditor
          id={id}
          wrapperId={wrapperId}
          labelText="Description"
          formik={formik}
          isDisabled={true}
          isRequired={true}
        ></RichTextEditor>
      </Provider>
    );
  });

  it('should render as expected', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('execute events', () => {
    findById(wrapper, eleWrapperId).simulate('blur');
    findById(wrapper, eleWrapperId).simulate('click');
    findById(wrapper, 'showmoreless-description').simulate('click');
  });

  it('Should set text color', () => {
    const textarea = wrapper.find('.public-DraftEditor-content').last();
    textarea.simulate('focus');
    const colorPickerIcon = wrapper.find('.text-color-picker');
    expect(colorPickerIcon.length).toBeGreaterThanOrEqual(1);

    colorPickerIcon.last().simulate('click');
    const colorPickerModal = wrapper.find('.rdw-colorpicker-modal.color');
    expect(colorPickerModal.length).toEqual(1);
    const colorPickerOption = colorPickerModal
      .find('.rdw-colorpicker-option')
      .first(); // #E25041
    colorPickerOption.simulate('click');

    findById(wrapper, eleWrapperId).simulate('blur');
  });

  it('Should set backgroud color', () => {
    const textarea = wrapper.find('.public-DraftEditor-content').last();
    textarea.simulate('focus');
    const bgcolorPickerIcon = wrapper.find('.bg-color-picker');
    expect(bgcolorPickerIcon.length).toBeGreaterThanOrEqual(1);

    bgcolorPickerIcon.last().simulate('click');
    const colorPickerModal = wrapper.find('.rdw-colorpicker-modal.bgcolor');
    expect(colorPickerModal.length).toEqual(1);
    const colorPickerOption = colorPickerModal
      .find('.rdw-colorpicker-option')
      .first(); // #E25041
    colorPickerOption.simulate('click');

    findById(wrapper, eleWrapperId).simulate('blur');
  });

  it('Should not close color picker modal dialog when switching between color and bgcolor', () => {
    const textarea = wrapper.find('.public-DraftEditor-content').last();
    textarea.simulate('focus');

    const colorPickerIcon = wrapper.find('.text-color-picker').last();
    const bgcolorPickerIcon = wrapper.find('.bg-color-picker').last();

    colorPickerIcon.simulate('click');
    expect(
      wrapper.find('.rdw-colorpicker-modal.color').length
    ).toBeGreaterThanOrEqual(1);

    bgcolorPickerIcon.simulate('click');
    expect(
      wrapper.find('.rdw-colorpicker-modal.bgcolor').length
    ).toBeGreaterThanOrEqual(1);

    colorPickerIcon.simulate('click');
    expect(
      wrapper.find('.rdw-colorpicker-modal.color').length
    ).toBeGreaterThanOrEqual(1);

    findById(wrapper, eleWrapperId).simulate('blur');
  });
});
