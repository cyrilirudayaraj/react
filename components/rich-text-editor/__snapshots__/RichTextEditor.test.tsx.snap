// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <RichTextEditor> should render as expected 1`] = `
"<Provider store={{...}}>
  <Connect(RichTextEditor) id=\\"description\\" wrapperId={1} labelText=\\"Description\\" formik={{...}} isDisabled={true} isRequired={true}>
    <RichTextEditor id=\\"description\\" wrapperId={1} labelText=\\"Description\\" formik={{...}} isDisabled={true} isRequired={true} usersInfo={{...}} fetchUsersOnce={[Function (anonymous)]} maxLength={4000} className=\\"fe_c_textarea fe_c_textarea--auto-size editor-wrapper\\" enableShowMore={true}>
      <div className=\\"textarea-component\\">
        <FormField id=\\"description\\" wrapperId={1} labelText=\\"Description\\" labelWidth={1} className=\\"rte\\" inputAs={[Function: r]} required={true} wrapperClassName=\\"fe_c_textarea fe_c_textarea--auto-size editor-wrapper fe_is-required fe_is-disabled\\" editorClassName=\\"editor-textarea\\" editorState={{...}} onEditorStateChange={[Function (anonymous)]} readOnly={true} wrapperStyle={{...}} toolbarHidden={true} error={[undefined]} toolbar={{...}} onChange={[Function (anonymous)]} onFocus={[Function (anonymous)]} onBlur={[Function (anonymous)]}>
          <FormFieldLayout labelWidth={1} nested={[undefined]} layout={[undefined]} statusExtras={[undefined]} className=\\"rte\\" labelAlwaysAbove={[undefined]} labelSlot={{...}} messageSlot={[undefined]} statusIndicatorSlot={[undefined]} inputSlot={{...}}>
            <div className=\\"fe_c_form-field fe_c_form-field--medium rte\\">
              <GridRow nested={true}>
                <div className=\\"fe_l_grid-row fe_l_grid-row--nested\\">
                  <GridCol width={{...}} className=\\"fe_c_form-field__label\\">
                    <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-1 fe_c_form-field__label\\">
                      <Label id=\\"description-label\\" htmlFor=\\"description\\" text=\\"Description\\" disabled={[undefined]} required={true} useLegend={false}>
                        <label htmlFor=\\"description\\" id=\\"description-label\\" className=\\"fe_c_label\\">
                          Description
                          <span className=\\"fe_c_label__required\\" aria-hidden=\\"true\\">
                             
                            - Required
                          </span>
                        </label>
                      </Label>
                    </div>
                  </GridCol>
                  <GridCol width={{...}} className=\\"fe_c_form-field__right\\">
                    <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-11 fe_c_form-field__right\\">
                      <div className=\\"fe_c_form-field__input-slot\\">
                        <r wrapperId={1} wrapperClassName=\\"fe_c_textarea fe_c_textarea--auto-size editor-wrapper fe_is-required fe_is-disabled\\" editorClassName=\\"editor-textarea\\" editorState={{...}} onEditorStateChange={[Function (anonymous)]} readOnly={true} wrapperStyle={{...}} toolbarHidden={true} toolbar={{...}} onChange={[Function (anonymous)]} onFocus={[Function (anonymous)]} onBlur={[Function (anonymous)]} className=\\"fe_c_form-field__input\\" id=\\"description\\" name=\\"description\\" required={true} hideRequiredStyles={[undefined]} disabled={[undefined]} error={[undefined]} aria-labelledby=\\"description-label\\" toolbarOnFocus={false} stripPastedStyles={false} localization={{...}} customDecorators={{...}}>
                          <div id=\\"rdw-wrapper-1\\" className=\\"fe_c_textarea fe_c_textarea--auto-size editor-wrapper fe_is-required fe_is-disabled rdw-editor-wrapper\\" style={{...}} onClick={[Function (anonymous)]} onBlur={[Function (anonymous)]} aria-label=\\"rdw-wrapper\\">
                            <div className=\\"editor-textarea rdw-editor-main\\" style={[undefined]} onClick={[Function (anonymous)]} onFocus={[Function (anonymous)]} onBlur={[Function (anonymous)]} onKeyDown={[Function: onKeyDown]} onMouseDown={[Function (anonymous)]}>
                              <DraftEditor keyBindingFn={[Function (anonymous)]} editorState={{...}} onChange={[Function (anonymous)]} blockStyleFn={[Function: D]} customStyleMap={{...}} handleReturn={[Function (anonymous)]} handlePastedText={[Function (anonymous)]} blockRendererFn={[Function (anonymous)]} handleKeyCommand={[Function (anonymous)]} ariaLabel=\\"rdw-editor\\" blockRenderMap={{...}} wrapperId={1} readOnly={true} className=\\"fe_c_form-field__input\\" id=\\"description\\" name=\\"description\\" required={true} hideRequiredStyles={[undefined]} disabled={[undefined]} error={[undefined]} aria-labelledby=\\"description-label\\" stripPastedStyles={false} ariaDescribedBy=\\"{{editor_id_placeholder}}\\" spellCheck={false}>
                                <div className=\\"DraftEditor-root\\">
                                  <div className=\\"DraftEditor-editorContainer\\">
                                    <div aria-activedescendant={{...}} aria-autocomplete={{...}} aria-controls={{...}} aria-describedby={[undefined]} aria-expanded={{...}} aria-label=\\"rdw-editor\\" aria-labelledby={[undefined]} aria-multiline={[undefined]} aria-owns={{...}} autoCapitalize={[undefined]} autoComplete={[undefined]} autoCorrect={[undefined]} className=\\"public-DraftEditor-content\\" contentEditable={false} data-testid={[undefined]} onBeforeInput={[Function (anonymous)]} onBlur={[Function (anonymous)]} onCompositionEnd={[Function (anonymous)]} onCompositionStart={[Function (anonymous)]} onCopy={[Function (anonymous)]} onCut={[Function (anonymous)]} onDragEnd={[Function (anonymous)]} onDragEnter={[Function (anonymous)]} onDragLeave={[Function (anonymous)]} onDragOver={[Function (anonymous)]} onDragStart={[Function (anonymous)]} onDrop={[Function (anonymous)]} onFocus={[Function (anonymous)]} onInput={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onKeyPress={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onPaste={[Function (anonymous)]} onSelect={[Function (anonymous)]} role={{...}} spellCheck={false} style={{...}} suppressContentEditableWarning={true} tabIndex={[undefined]}>
                                      <UpdateDraftEditorFlags editor={{...}} editorState={{...}} />
                                      <DraftEditorContents blockRenderMap={{...}} blockRendererFn={[Function (anonymous)]} blockStyleFn={[Function: D]} customStyleMap={{...}} customStyleFn={[undefined]} editorKey=\\"123\\" editorState={{...}} preventScroll={[undefined]} textDirectionality={[undefined]}>
                                        <div data-contents=\\"true\\">
                                          <div className=\\"\\" data-block={true} data-editor=\\"123\\" data-offset-key=\\"123-0-0\\">
                                            <DraftEditorBlock contentState={{...}} block={{...}} blockProps={[undefined]} blockStyleFn={[Function: D]} customStyleMap={{...}} customStyleFn={[undefined]} decorator={{...}} direction=\\"LTR\\" forceSelection={false} offsetKey=\\"123-0-0\\" preventScroll={[undefined]} selection={{...}} tree={{...}}>
                                              <div data-offset-key=\\"123-0-0\\" className=\\"public-DraftStyleDefault-block public-DraftStyleDefault-ltr\\">
                                                <DraftEditorLeaf offsetKey=\\"123-0-0\\" block={{...}} start={0} selection={{...}} forceSelection={false} text=\\"Testing value\\" styleSet={{...}} customStyleMap={{...}} customStyleFn={[undefined]} isLast={true}>
                                                  <span data-offset-key=\\"123-0-0\\" style={{...}}>
                                                    <DraftEditorTextNode>
                                                      <span data-text=\\"true\\">
                                                        Testing value
                                                      </span>
                                                    </DraftEditorTextNode>
                                                  </span>
                                                </DraftEditorLeaf>
                                              </div>
                                            </DraftEditorBlock>
                                          </div>
                                        </div>
                                      </DraftEditorContents>
                                    </div>
                                  </div>
                                </div>
                              </DraftEditor>
                            </div>
                          </div>
                        </r>
                      </div>
                      <div className=\\"fe_c_form-field__status\\" />
                    </div>
                  </GridCol>
                </div>
              </GridRow>
            </div>
          </FormFieldLayout>
        </FormField>
        <div className=\\"counter\\">
          <a id=\\"showmoreless-description\\" onClick={[Function: onClick]}>
            Show more
          </a>
        </div>
      </div>
    </RichTextEditor>
  </Connect(RichTextEditor)>
</Provider>"
`;
